### ISSUE TITLE
Would not update

### ISSUE THREAD

**Issue by JimF42** [synthetos/TinyG-Updater#1]
OK, I am doing this from memory... If you need more concrete details, like the exact messages, I will get them, but I think this failed in such a way that will not be required.

I downloaded the TinyG-Updater and ran it from my laptop that has the TinyG connected to it. It is a Windows 7 computer and the TinyG connects as COM3.

After selecting the Edge release and leaving the "Attempt automatic reset" checked and doing a "Lookup" for my device, I attempted the update. The status window came up, briefly said Connecting, then very quickly the progress bar turned red and said (from memory) that the _verification failed_.  The Console Log window is **completely empty**--not a single line in it.

I tried this several times... With the checkbox checked, with it unchecked and me manually doing the reset on the board... all to no avail. I even manually placed the .hex file in the same folder incase that was an issue. Nope.

Next I will tell you how I got my TinyG updated in case it has any bearing on why this tool failed for me.

Then, I decided to switch to using the prior method of using avrdude. I read and followed the instructions at https://github.com/synthetos/TinyG/wiki/TinyG-Updating-Firmware. In step 3 I was directed to install the "Arduino IDE". I decided to just download the .zip file and not the installer because I figured I only needed the one file. When I tried running avrdude, a dialog box popped up and told me that it could not find the cygwin1.dll (probably because I didn't do an install). Running short on time, I just copied all of the DLLs in the root of the zip to the same folder as avrdude.

After adding a parameter so it could find the config file, avrdude was finally running... But, I was getting tons of **"***failed"** lines... Looking at the troubleshooting section, it said to try adding **-e** and says _"This option erases a Flash page before writing it and in some cases it's required."_

Yup, it was required for me. After that my TinyG would finally updated!!!

So, hopefully my description of how I got my TinyG updated will help in getting this GUI to work. 

If you need more information, please feel free to ask.

Thanks,
Jim

**Comment by ril3y** [comment]
Yes I have seen this too... I have had to use the -e erase flag before
too.  Hmm I thought that was getting put in to the updater.  I will check.

**Comment by JimF42** [comment]
Any fixes for this yet? I need to do another update and this still does not work for me. I will revert to avrdude again.

Thanks

**Comment by JimF42** [comment]
I just downloaded a fresh copy and it still does not work. Am I the only one using this that is having this problem? Is there something else I can do to get this to work instead of using avrdude? Is it an issue with my computer or with my TinyG board?

**Comment by ril3y** [comment]
No you are not.  This is an ISSUE.  I am trying to get to the bottom of
it.  You are on windows right?  Right now you can use the avrdude
commandline option to upgrade.  Carl did I post on it.  I will link him
in.  I cant find it at the moment hopefully he will chime in.

Riley

**Comment by ril3y** [comment]
Try avrdude here: 
https://groups.google.com/forum/#!topic/devtinyg/B9xSJaupzNQ

**Comment by JimF42** [comment]
Yes, I am on Windows and yes, I got the firmware updated via avrdude. Thanks for the info.

**Comment by JimF42** [comment]
I just took a quick look at the project's source tree and noticed (if I understand it correctly without studying it) that you are including avrdude in your code. Is it not possible to just add the "-e" option when you run it? Does adding it break something for others?

**Comment by jkanzaki** [comment]
Hi Riley 

The link you provided is good one, I am not a computer wiz and using command line is new to me. Something to help other who doesn't know how to use command line to navigate to the directory could look into this link 
http://www.pcstats.com/articleview.cfm?articleid=1723&page=3

simple stuff like change directory "cd" or cd.. etc.. 

Everything else is self explanatory. 

Thanks again for posting the info. 
Henry

**Comment by sraney** [comment]
I am running into the same issue with using the Synthetos TinyG Updater.  I'll add bit more information to this.  

System Information: 

```
OS:  Windows 8.1 64bit 
Ram: 4G (3.88 usable) 
Com port: COM3
```

TinyG Information

```
SN: Ask if needed
Ver v8
Firmware:  435.1 (reported by Synthetos TinyG Updater) 
```

I've tried a Windows 7 system and it seems to fail also, but I have to admit, it always seems to choke on atom xplatform items.  Electron works though.  

Tested paths; all fail: 
- Tested with firewall and AV off 
- Tested with System Administrator permissions 
-  Tested from with a direct call to the application via commandline with the following via the command line: 

```
c:\Archives\TinyG-Updater-v0.5-Windows>[2908:0722/181906:INFO:CONSOLE(0)] Node v
ersion: v0.11.13
[5676:0722/181906:INFO:renderer_main.cc(227)] Renderer process started
[2908:0722/181907:WARNING:dhcp_proxy_script_adapter_fetcher_win.cc(264)] Error f
etching PAC URL from DHCP: 2
[2908:0722/181908:INFO:CONSOLE(0)] requested load-hex: tinyg-master-440.16.hex
[2908:0722/181908:INFO:CONSOLE(0)] requested load-hex: tinyg-edge-440.14.hex
[2908:0722/181908:INFO:CONSOLE(0)] Check SHA1: fab193605c00ed7c7398ffb97c377fc96
c0065a1
[2908:0722/181916:WARNING:dhcp_proxy_script_adapter_fetcher_win.cc(264)] Error f
etching PAC URL from DHCP: 2
[2908:0722/181921:INFO:CONSOLE(0)] tg> {"r":{"fb":435.10},"f":[1,0,12,2672]}
[2908:0722/181921:INFO:CONSOLE(0)] fb: 435.1
[2908:0722/181921:INFO:CONSOLE(0)] closed> undefined
```

along with this error that always pops up 

![synthetos_updater_error_07_22_2015](https://cloud.githubusercontent.com/assets/9012610/8841118/bfef7aa2-309e-11e5-85ce-90b5e59e0aa4.png)

**Comment by DeKevin** [comment]
Hi guys,

I had the same issue as JimF42. The app worked for me after I put in a "libusb0.dll" file in the bin folder where the app is installed, something like this:

C:\TinyG-Updater-v0.5-Windows\resources\app\bin

**Comment by giseburt** [comment]
Ok, we just released v0.6.1 of the Updater. It should fix all known issues. Please let us know if they still exist.

_Update:_ Link: https://github.com/synthetos/TinyG-Updater/releases/tag/v0.6.1
